File: app/src/main/java/ru/example/newsreader/room/Dao/SourceDao.java
package ru.example.newsreader.room.Dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import ru.example.newsreader.room.Entity.SourceEntity;

@Dao
public interface SourceDao {
    @Insert
    void insert(SourceEntity sourceEntity);

    @Update
    void update(SourceEntity sourceEntity);

    @Delete
    void delete(SourceEntity sourceEntity);

    @Query("SELECT * FROM SourceEntity")
    List<SourceEntity> getAllSources();

    @Query("DELETE FROM SourceEntity")
    void deleteAll();


}

File: app/src/main/java/ru/example/newsreader/room/Dao/SourceDao.kt
package ru.example.newsreader.room.Dao

import android.arch.persistence.room.*
import ru.example.newsreader.room.Entity.SourceEntity

@Dao
interface SourceDao {
    @Insert
    fun insert(sourceEntity: SourceEntity)

    @Update
    fun update(sourceEntity: SourceEntity)

    @Delete
    fun delete(sourceEntity: SourceEntity)

    @Query("SELECT * FROM SourceEntity")
    fun getAllSources() : List<SourceEntity>

    @Query("DELETE FROM SourceEntity")
    fun deleteAll()
}